assembly/src/library/mod.rs

```diff
@@ -41,6 +41,9 @@ pub trait Library {
 
     /// Returns the dependency libraries of this library.
     fn dependencies(&self) -> &[LibraryNamespace];
+
+    /// Returns the AST of the module stored at the provided path.
+    fn get_module_ast(&self, path: &LibraryPath) -> Option<ModuleAst>;
```

Comment:
Can we return `Option<&ModuleAst>` from here? This way, we can avoid cloning modules (which may be a bit expensive).

Also, should we provide a default implementation of this method - i.e., something like:
```Rust
self.modules()
    .iter()
    .find(|&module| module.path == *path)
    .map(|module| module.ast)
```